### ISSUE TITLE
Sometimes "git pull" errors with "fatal: Cannot rebase onto multiple branches." but is fine immediately afterwards

### ISSUE THREAD

**Issue by tomchiverton** [magicmonty/bash-git-prompt#523]
```
$ cd x.y
~/workspace/docker/sites/x.y [master|…13⚑ 1] 
09:41 $ git pull
fatal: Cannot rebase onto multiple branches.
~/workspace/docker/sites/x.y [master ↓·18|…13⚑ 1] 
09:41 $ git pull
Updating d1f83acf0..ae2ab08ef
Fast-forward
```

I assume something is going on in the background, and the prompt isn't waiting ?

**Comment by ogr3** [comment]
yes, that's the problem with the prompt updating in the background. Its mostly nice but this can happen - just wait a second or two and re-issue the pull command.

This happens in the git plugin for oh-my-zsh as well

**Comment by tomchiverton** [comment]
Is there no way to run the update synchronously ? I didn't see an option, and "git fetch" is normally quick...

**Comment by ogr3** [comment]
Not configurable at the moment - it was introduced 10 tears ago as a key feature of this project in #23

**Comment by tomchiverton** [comment]
What about an extra delay before stating the background task ? This would give a chance for a "git X" after a "cd" into a clone to run first ?

**Comment by cnmcavoy** [comment]
This annoys me too...

I added this monstrosity to my bashrc to work around it. Would be interested in improvements or other alternative suggestions:
```
function safe_git_pull {
  while true; do
    if test -f ".git/index.lock"; then
      sleep 0.1
    else
      /usr/bin/git pull $@
      return $?
    fi
  done
}

function git {
  if [[ "$1" == "pull" ]]; then
   shift;
   safe_git_pull $@
  else
    /usr/bin/git $@
  fi
}
```

**Comment by rakotomandimby** [comment]
Hello,

In the issue https://github.com/magicmonty/bash-git-prompt/issues/517, the maintainer has announced his retirement from this project.

I forked it on https://github.com/rakotomandimby/bash-git-prompt, and if this issue is still relevant, I kindly ask you to re-create it over there, and provide the pull request over there.

Thank you
